ctapipe/io/dstio.py

```diff
@@ -0,0 +1,266 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Components to read HESSIO DST data.  
+
+This requires the hessio python library to be installed
+"""
+import logging
+
+from .containers import DataContainer
+from ..core import Provenance
+
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+import numpy as np
+from ctapipe.calib.camera import DstioR1Calibrator
+
+logger = logging.getLogger(__name__)
+
+try:
+    from pyhessio import open_hessio
+    from pyhessio import HessioError
+    from pyhessio import HessioTelescopeIndexError
+    from pyhessio import HessioGeneralError
+except ImportError as err:
+    logger.fatal(
+        "the `pyhessio` python module is required to access MC data: {}"
+        .format(err))
+    raise err
+
+__all__ = [
+    'dst_event_source',
+]
+
+
+def dst_get_list_event_ids(url, max_events=None):
+    """
+    Faster method to get a list of all the event ids in the dst file.
+    This list can also be used to find out the number of events that exist
+    in the file.
+
+    Parameters
+    ----------
+    url : str
+        path to file to open
+    max_events : int, optional
+        maximum number of events to read
+
+    Returns
+    -------
+    event_id_list : list[num_events]
+        A list with all the event ids that are in the file.
+
+    """
+    logger.warning("This method is slow. Need to find faster method.")
```

Comment:
This is a function we cannot guarantee will work anyhow, since it only works if the file is fully available and is not a stream. Even on the cluster, we may use some form of streaming IO like XROOTD, or similar, so may not always be a way to build such a list.   In other file format readers, this is referred to as building an index table, so we might want to rename the function to something like "build_event_index", and note that it only works if it is possible to loop multiple times over the input, and should not be relied upon.   In fact, in some formats like HDF5 (which we'll probably use for intermediate outputs), such an index can be built automatically by the hdf5 library.